File: src/com/campaignio/tasklets/sms/Condition.java
package com.campaignio.tasklets.sms;

import org.json.JSONObject;

import com.campaignio.tasklets.TaskletAdapter;
import com.campaignio.tasklets.TaskletManager;

public class Condition extends TaskletAdapter
{
    // Fields
    public static String IF_TYPE = "if_type";
    public static String IF_TYPE_VALUE = "value";
    public static String IF_TYPE_STRLEN = "strlen";

    // Comparator
    public static String COMPARATOR = "comparator";
    public static String COMPARATOR_LESS_THAN = "less_than";
    public static String COMPARATOR_GREATER_THAN = "greater_than";
    public static String COMPARATOR_LESS_THAN_OR_EQUAL_TO = "less_than_or_equals";
    public static String COMPARATOR_GREATER_THAN_OR_EQUAL_TO = "greater_than_or_equals";
    public static String COMPARATOR_NOT_EQUAL_TO = "not_equal_to";
    public static String COMPARATOR_EQUAL_TO = "equal_to";

    // Variable 1 and 2
    public static String VARIABLE_1 = "variable_1";
    public static String VARIABLE_2 = "variable_2";

    // Branches - Yes/No
    public static String BRANCH_YES = "Yes";
    public static String BRANCH_NO = "No";

    // Run
    public void run(JSONObject campaignJSON, JSONObject subscriberJSON,
	    JSONObject data, JSONObject nodeJSON) throws Exception
    {

	// Get Variables
	String variable1 = getStringValue(nodeJSON, subscriberJSON, data,
		VARIABLE_1);
	String variable2 = getStringValue(nodeJSON, subscriberJSON, data,
		VARIABLE_2);

	// Trim the variables
	if (variable1 != null)
	    variable1 = variable1.trim();
	if (variable2 != null)
	    variable2 = variable2.trim();

	// If Type
	String ifType = getStringValue(nodeJSON, subscriberJSON, data, IF_TYPE);
	if (ifType.equalsIgnoreCase(IF_TYPE_STRLEN))
	    variable1 = variable1.length() + "";

	String branch = BRANCH_NO;

	String comparator = getStringValue(nodeJSON, subscriberJSON, data,
		COMPARATOR);

	System.out.println("Variable 1: " + variable1 + " " + variable2 + " "
		+ comparator);

	try
	{
	    // Do the required operation. If matches send to yes
	    if (comparator.equalsIgnoreCase(COMPARATOR_LESS_THAN))
	    {
		// Convert the string into Integer and check
		if (Long.parseLong(variable1) < Long.parseLong(variable2))
		    branch = "yes";
	    }

	    if (comparator.equalsIgnoreCase(COMPARATOR_LESS_THAN_OR_EQUAL_TO))
	    {
		// Convert the string into Integer and check
		if (Long.parseLong(variable1) <= Long.parseLong(variable2))
		    branch = "yes";
	    }

	    if (comparator.equalsIgnoreCase(COMPARATOR_GREATER_THAN))
	    {
		if (Long.parseLong(variable1) > Long.parseLong(variable2))
		    branch = "yes";
	    }

	    if (comparator
		    .equalsIgnoreCase(COMPARATOR_GREATER_THAN_OR_EQUAL_TO))
	    {
		if (Long.parseLong(variable1) >= Long.parseLong(variable2))
		    branch = "yes";
	    }

	    if (comparator.equalsIgnoreCase(COMPARATOR_GREATER_THAN))
	    {
		if (Long.parseLong(variable1) > Long.parseLong(variable2))
		    branch = "yes";
	    }

	    if (comparator
		    .equalsIgnoreCase(COMPARATOR_GREATER_THAN_OR_EQUAL_TO))
	    {
		if (Long.parseLong(variable1) >= Long.parseLong(variable2))
		    branch = "yes";
	    }

	    if (comparator.equalsIgnoreCase(COMPARATOR_NOT_EQUAL_TO))
	    {
		if (!variable1.equalsIgnoreCase(variable2))
		    branch = "yes";
	    }

	    if (comparator.equalsIgnoreCase(COMPARATOR_EQUAL_TO))
	    {
		if (variable1.equalsIgnoreCase(variable2))
		    branch = "yes";
	    }
	}
	catch (Exception e)
	{
	    e.printStackTrace();
	}

	// Go to next tasks
	TaskletManager.executeTasklet(campaignJSON, subscriberJSON, data,
		nodeJSON, branch);

    }

}

File: war/min/json/nodes/images/smsnodes/condition.js
{
    "name": "Condition",
    "thumbnail": "json/nodes/images/sms/Condition.png",
    "icon": "json/nodes/icons/sms/Condition.png",
    "info": "This will check answer given by contestant is currect or not.",
    "help": "This will check answer given by contestant is currect or not.",
    "author": "john",
    "company": "Invox",
    "language": "en",
    "branches": "No,Yes",
    "workflow_tasklet_class_name": "com.campaignio.tasklets.sms.Condition",
    "category": "Voice Automation-IVR, Developers",
    "ui": [
        {
            "label": "IF",
            "required": "required",
            "category": "Settings",
            "name": "if_type",
            "title": "Variable type.",
            "multiple": "multiple",
            "ismultiple": "true",
            "value": "value",
            "options": {
                "value": "value",
                "strlen": "strlen"
            },
            "fieldType": "select",
            "type": "select" 
        },
        {
            "label": "of",
            "required": "required",
            "category": "Settings",
            "value": "",
            "name": "variable_1",
            "title": "Variable name (eg: $variable) or value (eg: 5).",
            "fieldType": "input",
            "type": "text" 
        },
        {
            "label": "compare this",
            "required": "required",
            "category": "Settings",
            "name": "comparator",
            "title": "Operation type.",
            "multiple": "multiple",
            "ismultiple": "true",
            "value": "<",
            "options": {
                "less than": "less_than",
                "greater than": "greater_than",
                "less than or equals": "less_than_or_equals",
                "greater than or equals": "greater_than_or_equals",
                "not equal to": "not_equal_to",
                "equal to": "equal_to"
            },
            "fieldType": "select",
            "type": "select" 
        },
        {
            "label": "with",
            "required": "required",
            "category": "Settings",
            "name": "variable_2",
            "title": "Variable name (eg: $variable) or value (eg: 5).",
            "fieldType": "input",
            "type": "text" 
        },
        {
            "label": "Invox lets you check conditions between a literal and another literal -- or between a literal and a constant. (The conditions are: greater than, less than, equal to, and not equal to.) <br><br>There is no need to add $ before the variable name. For a constant, please enclose your value in quotes () if it is a string.<br><br>Strlen (String length) allows you to check for empty variables (When you are retrieving data from the database or REST, variables may be empty). <br><br>This is also useful when checking the length of a variable from Get Input if you want to restrict your callers to entering a fixed number of digits from the keypad.</br>",
            "category": "Help",
            "fieldType": "label",
            "type": "label" 
        } 
    ]
}